acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -7,138 +7,413 @@ use serde::{Deserialize, Serialize};
+                    })
+                }
+            }
+            BlackBoxFunc::FixedBaseScalarMul => {
+                if inputs.is_empty() {
+                    Err(std::io::ErrorKind::InvalidData.into())
+                } else {
+                    Ok(BlackBoxFuncCall::FixedBaseScalarMul { input: inputs[0], outputs })
+                }
+            }
+            BlackBoxFunc::Keccak256 => Ok(BlackBoxFuncCall::Keccak256 { inputs, outputs }),
+        }
+    }
+}
+
+const ABBREVIATION_LIMIT: usize = 5;
+
+fn get_inputs_string(inputs: &[FunctionInput]) -> String {
+    // Once a vectors length gets above this limit,
+    // instead of listing all of their elements, we use ellipses
+    // to abbreviate them
+    let should_abbreviate_inputs = inputs.len() <= ABBREVIATION_LIMIT;
+
+    if should_abbreviate_inputs {
+        let mut result = String::new();
+        for (index, inp) in inputs.iter().enumerate() {
+            result += &format!("(_{}, num_bits: {})", inp.witness.witness_index(), inp.num_bits);
+            // Add a comma, unless it is the last entry
+            if index != inputs.len() - 1 {
+                result += ", "
+            }
         }
+        result
+    } else {
+        let first = inputs.first().unwrap();
+        let last = inputs.last().unwrap();
+
+        let mut result = String::new();
+
+        result += &format!(
+            "(_{}, num_bits: {})...(_{}, num_bits: {})",
+            first.witness.witness_index(),
+            first.num_bits,
+            last.witness.witness_index(),
+            last.num_bits,
+        );
+
+        result
+    }
+}
+
+fn get_outputs_string(outputs: &[Witness]) -> String {
+    let should_abbreviate_outputs = outputs.len() <= ABBREVIATION_LIMIT;
 
-        let num_outputs = read_u32(&mut reader)?;
-        let mut outputs = Vec::with_capacity(num_outputs as usize);
-        for _ in 0..num_outputs {
-            let witness = Witness(read_u32(&mut reader)?);
-            outputs.push(witness)
+    if should_abbreviate_outputs {
+        let mut result = String::new();
+        for (index, output) in outputs.iter().enumerate() {
+            result += &format!("_{}", output.witness_index());
+            // Add a comma, unless it is the last entry
+            if index != outputs.len() - 1 {
+                result += ", "
+            }
         }
+        result
+    } else {
+        let first = outputs.first().unwrap();
+        let last = outputs.last().unwrap();
 
-        Ok(BlackBoxFuncCall { name, inputs, outputs })
+        let mut result = String::new();
+        result += &format!("(_{},...,_{})", first.witness_index(), last.witness_index());
+        result
     }
 }
 
 impl std::fmt::Display for BlackBoxFuncCall {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let uppercase_name: String = self.name.name().into();
+        let uppercase_name: String = self.get_black_box_func().name().into();
```

Comment:
This seems to be done a lot, why not just have a `.name()` function on `BlackBoxFuncCall`?